### PR TITLE
Test across multiple versions of _ and bb and mn

### PR DESCRIPTION
**Pull request by samccone**

### CODE REVIEW

**Comment by jasonLaster** [comment]
travis woes
```
lodash@2.4.1 node_modules/lodash
mv: cannot move `node_modules/lodash/lodash.js' to `node_modules/underscore/underscore.js': No such file or directory
The command "curl -L https://rawgit.com/marionettejs/travis-ci-setup/master/ci.sh | bash" failed and exited with 1 during .
```

**Comment by samccone** [comment]
grrrr

**Comment by paulfalgout** [comment]
So I fixed the lodash issue, but now a test is failing.  However this test passes for me locally with lodash 2.4.1:

```
lodash@2.4.1 node_modules/lodash
sh-3.2# npm test

> marionette.inspector@0.5.3 test /Users/paulfalgout/marionette.inspector
> grunt test --mocha-reporter=Spec;

Running "preprocess:core" (preprocess) task

Running "preprocess:agent" (preprocess) task

Running "preprocess:localAgent" (preprocess) task

Running "mocha:test" (mocha) task
Testing: extension/js/test/unit/AgentSpecRunner.html


  regionInspector
    showing a view
      ✓ the view is in the region tree 
      ✓ the view can be serialized in the tree 
      ✓ can use a subpath 
    showing a layout
      ✓ the layout is in the region tree 
      ✓ the layout's regions are in the region tree 
    showing a collection view
      ✓ the collectionView is in the region tree 
      ✓ the children views are in tree 

  serializeView
    ✓ serializes cid 
    ✓ serializes options 
    ✓ serializes ui 
    ✓ serializes element 
    ✓ serializes events 
```

**Comment by jasonLaster** [comment]
@paulfalgout which test is it? 

If were that close on getting this in, i'd do anything we need to figure this out including removing lodash support or the test :/  

It is weird that it passes locally though... were you using the curl lodash | npm test technique?

**Comment by paulfalgout** [comment]
I did the curl npm test thing.. the code I posted above was actually from the wrong test but `export LODASH=3.0` works for me locally but breaks on travis.  even tried matching the backbone versions..  works fine locally.

Here's where travis fails:
```
serializeView
    ◦ serializes cid:   ✓ serializes cid 
    ◦ serializes options:   ✓ serializes options 
    ◦ serializes ui:   1) "after each" hook
  9 passing (109ms)
  1 failing
  1)  "after each" hook:
     Uncaught TypeError: 'undefined' is not an object (evaluating 'Agent.lazyWorker.push') (file:///home/travis/build/marionettejs/marionette.inspector/extension/js/agent/build/src/core.js:30362)
```

**Comment by paulfalgout** [comment]
hmm it is in the afterEach... might just be something in the test itself.. but suuuper weird it works locally.

**Comment by paulfalgout** [comment]
Ok this exact same error just occurred on a previously passing underscore / backbone test run.

It looks like this issue is not lodash specific, but perhaps perf specific.

Backbone 1.0 Underscore 1.4.4

same error; different test.

```
 ◦ {a: Backbone.Model}:   1) "after each" hook
  29 passing (1s)
  1 failing
  1)  "after each" hook:
     Uncaught TypeError: 'undefined' is not an object (evaluating 'Agent.lazyWorker.push') (file:///home/travis/build/marionettejs/marionette.inspector/extension/js/agent/build/src/core.js:29833)
```

**Comment by paulfalgout** [comment]
Ok.. so I don't think the issues we are seeing relate to any test specifically.

I _think_ [this](https://github.com/marionettejs/marionette.inspector/commit/ea5bf33b38de6cb2d5ba73d3a5fac068a5f13cb2#diff-d346c9fc3e2f89c13de812c2a0d80313L57) resolves the issues.  I don't know if there are ramifications for not deleting it here.  
 
@jasonLaster ?

**Comment by paulfalgout** [comment]
Another update.  Added multiple Mn 2.1, 2.2, 2.3, and 2.4 to the matrix.  All seems to work fine.

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "sinon": "1.10.3",
     "sinon-chai": "2.5.0",
     "bower": "1.3.12",
-    "grunt-sass": "0.17.0"
+    "grunt-sass": "0.17.0",
+    "underscore": "1.8.3"
   }
```
Comment (jasonLaster) [inline package.json]: We're not overriding underscore here are we?

Also how are these versions then used by the agent test runner?

Comment (samccone) [inline package.json]: in the ci script we simply move the index file :) it is a total hack but it works :)

Comment (paulfalgout) [inline package.json]: underscore/lodash were not npm dependencies before, so we aren't overriding anything in use, but yeah the matrix test will override underscore with the various versions and with lodash.

**Comment by jasonLaster** [comment]
:boom: it's in

**Pull request merged**